Add warnings for empty gcov (#448)

stephematician opened the pull request:
Add and test warnings if gcov missing or empty output to address #448

Simple warnings produced when there are .gcno files, but either gcov not found (i.e. covr.gcov is empty) or gcov output is empty.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/r-lib/covr?pullRequest=465) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/r-lib/covr?pullRequest=465) before we can accept your contribution.<br/><hr/>**Stephen Wade** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/r-lib/covr?pullRequest=465) it.</sub>

jimhester commented:
Thanks! I think this would be better off as an error rather than a warning, could you make that change and could you also please add a bullet t NEWS? It should briefly [describe the change](https://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

stephematician commented:
Can do - but first I'd like to confirm. There are two different warnings I've inserted; should both be made into errors? Also I had a look at `icc.R` and similar checks there only result in warnings. Does this all make sense?

jimhester commented:
https://github.com/r-lib/covr/pull/465/files#diff-f748633604ad9005c853f36c4be958c6353e3ecb46bb4b3ee85bc4e7c3430669R73 should be an error I think.

stephematician commented:
Ok, yep that makes sense to me. I've switched the warning to an error (and updated corresponding test), + added to NEWS.md. Hopefully that covrs everything! ;) ... except for the pesky digest package that doesn't support 3.2

jimhester commented:
![Thanks a bunch!](https://media.giphy.com/media/3ohs4kI2X9r7O8ZtoA/giphy.gif)
